sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/CostBasedJoinReorder.scala

```diff
@@ -51,6 +51,11 @@ case class CostBasedJoinReorder(conf: CatalystConf) extends Rule[LogicalPlan] wi
 
   def reorder(plan: LogicalPlan, output: AttributeSet): LogicalPlan = {
     val (items, conditions) = extractInnerJoins(plan)
+    // Find the star schema joins. Currently, it returns the star join with the largest
+    // fact table. In the future, it can return more than one star join (e.g. F1-D1-D2
+    // and F2-D3-D4).
+    val starJoinPlans = StarSchemaDetection(conf).findStarJoins(items, conditions.toSeq)
```

Comment:
@gatorsmile @ioana-delaney Thank you for your replies.  My main point is to identify the deficiency of the DP algorithm so that we can make improvement.  Since you are familiar with DP algorithm, can you help us identify its deficiency/limitations so that we can improve it?

One deficiency the DP algorithm has is the explosion of the search space when there is a large  number of join relations such as >30.   In CostBasedJoinReorder, we do not optimize join order if the number of join relations is greater than the threshold value  joinReorderDPThreshold.  I think this is a place star join reorder algorithm can help.  This is because it defaults to left-deep tree which is a smaller search space.  What do you think?